### PR TITLE
New ThingHandlerService documentation

### PR DESCRIPTION
**Pull request by J-N-K**
Related to https://github.com/openhab/openhab-core/pull/3957

### CODE REVIEW

**Review by stefan-hoehn — APPROVED** [review]
LGTM, Jan.

File: developers/bindings/index.md [hunk 1]
```diff
@@ -976,41 +978,45 @@ Here the developer only needs to implement four simple methods:
...
-The handler can be used to get the bridge UID or to get access to the configured device connected to the bridge handler.
-
 In the bridge handler you need to activate the thing handler service.
 This is done by implementing the `getServices` method in your bridge handler:
 
 ```java
     @Override
     public Collection<Class<? extends ThingHandlerService>> getServices() {
-        return Collections.singleton(<your binding bridge DiscoveryService>.class);
+        return Collections.singleton(YourBindingDiscoveryService.class);
```
Comment (wborn) [inline developers/bindings/index.md:1019]:
```suggestion
        return List.of(YourBindingDiscoveryService.class);
```

**Comment by wborn** [comment]
Maybe you can also update the other 2 lines using `Collections.singleton` ? :slightly_smiling_face:

**Comment by stefan-hoehn** [comment]
Can someone give me a sign when we are done with the review? :)

File: developers/bindings/index.md [hunk 2]
```diff
@@ -976,41 +978,44 @@ Here the developer only needs to implement four simple methods:
...
+@Component(scope = ServiceScope.PROTOTYPE, service = YourBindingDiscoveryService.class)
+public class <YourBindingDiscoveryService> extends AbstractThingHandlerDiscoveryService<YourBridgeHandler> {
 ```
 
-The interface `ThingHandlerService` has 2 methods to pass the handler of the bridge.
-A typical implementation is:
+In the class there is a field `protected YourBridgeHandler thingHandler;` which is automatically assigned.
+This field is guaranteed to be non-null after the service has been activated.
+
+During service creation, first `activate()` is called, then the `thingHandler` is injected and finally ìnitialize() is called.
```
Comment (wborn) [inline developers/bindings/index.md:992]:
```suggestion
During service creation, first `activate()` is called, then the `thingHandler` is injected and finally `ìnitialize()` is called.
```

File: developers/bindings/index.md [hunk 3]
```diff
@@ -976,41 +978,44 @@ Here the developer only needs to implement four simple methods:
 When the discovery process is dependent on a configured bridge the discovery service must be bound to the bridge handler.
 Binding additional services to a handler can be achieved by implementing the service as a `ThingHandlerService`.
 
-Instead of using the Component annotation your discovery service implements the `ThingHandlerService`.
-It should extend the `AbstractDiscoveryService` (which implements `DiscoveryService`) just like a normal service:
+It should extend the `AbstractThingHandlerDiscoveryService` (which implements `ThingHandlerService`and `DiscoveryService`) just like a normal service.
```
Comment (wborn) [inline developers/bindings/index.md:981]:
```suggestion
It should extend the `AbstractThingHandlerDiscoveryService` (which implements `ThingHandlerService` and `DiscoveryService`) just like a normal service.
```

File: developers/bindings/index.md [hunk 4]
```diff
@@ -976,41 +978,44 @@ Here the developer only needs to implement four simple methods:
...
+public class <YourBindingDiscoveryService> extends AbstractThingHandlerDiscoveryService<YourBridgeHandler> {
 ```
 
-The interface `ThingHandlerService` has 2 methods to pass the handler of the bridge.
-A typical implementation is:
+In the class there is a field `protected YourBridgeHandler thingHandler;` which is automatically assigned.
+This field is guaranteed to be non-null after the service has been activated.
+
+During service creation, first `activate()` is called, then the `thingHandler` is injected and finally ìnitialize() is called.
+The opposite order is used when the service is destroyed, first `dispose()` is called (when the thing handler is still available in the service), then `deactivate`.
```
Comment (wborn) [inline developers/bindings/index.md:993]:
```suggestion
The opposite order is used when the service is destroyed, first `dispose()` is called (when the thing handler is still available in the service), then `deactivate()`.
```

**Review by wborn — COMMENTED** [review]
For me it's done after these minor comments have been addressed:

**Comment by J-N-K** [comment]
done

**Review by wborn — APPROVED** [review]
LGTM too now!

**Review by stefan-hoehn — APPROVED** [review]
2 me 2, thanks

**Pull request merged**